src/App.jsx

```diff
@@ -1,13 +1,21 @@
 import './App.css';
 import TodoList from './TodoList';
 import AddTodoForm from './AddTodoForm';
-import { useState } from 'react';
+import { useState, useEffect } from 'react';
+
+const useSemiPersistentState = (key, initialState) => {
+  const [value, setValue] = useState(JSON.parse(localStorage.getItem('savedTodoList')) || initialState);
+  useEffect(() => {
+    localStorage.setItem(key, JSON.stringify(value));
+  }, [value, key]);
+  return [value, setValue];
+};
 
 const App = () => {
+  const [todoList, setTodoList] = useSemiPersistentState('savedTodoList', []);
```

Comment:
`useSemiPersistentState` should have no arguments here. Try to incorporate `savedTodoList` and the initial state inside your function

Reply:
Thank you, makes sense, done.